File: prototype/go/prototype_test.go
package prototype

import "testing"

func TestPrototype(t *testing.T) {
	AddGamePlayer("1", &GamePlayer{1, "张三", &Blood{111}})
	AddGamePlayer("2", &GamePlayer{2, "李四", &Blood{222}})
	AddGamePlayer("3", &GamePlayer{3, "王五", &Blood{333}})

	oldPlayers := GetGamePlayers()
	RefreshAllGamePlayers()
	newPlayers := GetGamePlayers()
	// newPlayers["3"].Blood.Volume = 123
	t.Logf("原型模式: old players=>%s, new players=>%s, old==new?=>%t\n",
		oldPlayers, newPlayers, &oldPlayers == &newPlayers)
}

File: prototype/java/PrototypeTest.java
import java.util.Map;

public class PrototypeTest {
    public static void main(String[] args) {
        Prototype prototype = new Prototype();
        prototype.addGamePlayer("1", new GamePlayer(1, "张三", new Blood(111)));
        prototype.addGamePlayer("2", new GamePlayer(2, "李四", new Blood(222)));
        prototype.addGamePlayer("3", new GamePlayer(3, "王五", new Blood(333)));

        Map<String, GamePlayer> oldPlayers = prototype.gamePlayers();
        prototype.refreshAll();
        Map<String, GamePlayer> newPlayers = prototype.gamePlayers();
        // newPlayers.get("3").getBlood().setVolume(123);

        System.out.printf("原型模式: old players=>%s, new players=>%s, old==new?=>%b%n",
                oldPlayers, newPlayers, oldPlayers == newPlayers);
    }
}
